src/test/java/htsjdk/samtools/util/BlockCompressedInputStreamTest.java

```diff
@@ -86,4 +91,142 @@ public void available_should_return_number_of_bytes_left_in_current_block() thro
 			}
 		}
 	}
+
+    @Test
+    public void testCustomInflaterFileInput() throws Exception {
```

Comment:
I think that the three test cases you added here could be unified into a single test case via a `DataProvider` that supplies `BlockCompressedInputStreams` constructed in each of the three ways, as well as expected decompressed file contents.